plugin/settings/checkbox.php

```diff
@@ -13,28 +13,52 @@
  * This will become unnecessary when using the new AJAX/API/Form module, because
  * they use a smart JavaScript function to gather form fields data.
  */
-function render_setting_field_checkbox($config) {
+function render_setting_field_checkbox($config, $type='checkbox') {
 
   foreach ([
     'name',
     'value',
-    'label'
+    'label',
+    'description'
   ] as $key) {
     $$key = $config[$key];
```

Comment:
```suggestion
    $$key = $config[$key] ?? '';
```

This change will prevent "undefined property" warning for the newly added `description`, which existing uses of the function haven't defined. It makes it optional.